File: android/JavaHelloWorld/src/Calculator.java

public class Calculator {
    public static class DivisionByZeroException extends Exception {
        public DivisionByZeroException(String message) {
            super(message);
        }
    }

    public static double calc(double firstVal, ArithmeticAction action, double secondVal) throws DivisionByZeroException {
        switch (action.getType()) {
            case ADD:
                return firstVal + secondVal;
            case SUB:
                return firstVal - secondVal;
            case MULT:
                return firstVal * secondVal;
            case DIV:
                if(secondVal == 0) {
                    throw new DivisionByZeroException("Calculator: division by zero!");
                } else {
                    return firstVal / secondVal;
                }
            default:
                return 0;
        }
    }
}

File: android/KotlinHelloWorld/src/main/kotlin/Calculator.kt

class Calculator {
    class DivideByZeroException(message: String) : Exception(message)
    class WrongAction(message: String) : Exception(message)
    fun calc(first: Double, action: ArithmeticAction, second:Double): Double {
        when(action) {
            ArithmeticAction.Plus -> return first + second
            ArithmeticAction.Minus -> return first - second
            ArithmeticAction.Multiply -> return first * second
            ArithmeticAction.Divide -> {
                if(second == 0.0) {
                    throw DivideByZeroException("Calculator: division by zero!")
                } else {
                    return first / second
                }
            }
            else -> {
                throw WrongAction("Calculator: wrong arithmetic action!")
            }
        }
    }
}
